doc/weakly_supervised.rst

```diff
@@ -700,6 +700,60 @@ of triplets that have the right predicted ordering.
 Algorithms
 ----------
 
+.. _scml:
+
+:py:class:`SCML <metric_learn.SCML>`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Sparse Compositional Metric Learning
+(:py:class:`SCML <metric_learn.SCML>`)
+
+`SCML` learns an squared mahalanobis distance from triplet constraints by 
+optimizing sparse positive weights assigned to a set of :math:`K` locally discriminative 
+rank-one PSD bases. This can be formulated as an optimization problem with only :math:`K`
+parameters, that can be solved with an efficient stochastic composite scheme.
+
+The Mahalanobis Matrix :math:`M` is built from a basis set :math:`B = \{b_i\}_{i=\{1,...,K\}}`
+weighted by a :math:`K` dimensional vector :math:`w = \{w_i\}_{i=\{1,...,K\}}` as:
+
+.. math::
+
+    M = \sum_{i=1}^K w_i b_i b_i^T = B \cdot diag(w) \cdot B^T \quad w_i \geq 0
+
+Learning :math:`M` in this form makes it PSD by design, as it is a nonnegative sum of PSD matrices.
+The optimization problem of :math:`w` over the triplets constraints :math:`C` is formulated as a
+classic margin-based hinge loss function over the relative constrains, a regularization :math:`\ell_1`
+is added to yield an sparse representation. The formulation is the following:
```

Comment:
an --> a
sparse weights or sparse combination, instead of sparse representation?